source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/telemetry/TfsTelemetryHelper.java

```diff
@@ -75,22 +81,42 @@ public synchronized static TelemetryClient getTelemetryClient() {
     }
 
     public static void sendMetric(final String name, final double value) {
+        if (sendingTelemetryDisabled) {
```

Comment:
It appears [TelemetryConfiguration ](https://github.com/Microsoft/ApplicationInsights-Java/blob/master/core/src/main/java/com/microsoft/applicationinsights/TelemetryConfiguration.java)has a "disable tracking" configuration, maybe we can leverage that instead?

Its method to disable tracking:
```
    public void setTrackingIsDisabled(boolean disable) {
        trackingIsDisabled = disable;
    }
```